R/expect_linear_time.R

```diff
@@ -0,0 +1,18 @@
+#' Function to test if input algorithm has linear time complexity
+#'
+#' @title Linear Time Complexity Test function
+#'
+#' @param f A function which returns a data frame composed of timings and corresponding data sizes, ideally from asymptoticTimings. (can use other functions)
+#'
+#' @param ... Parameters for passed function 'f'.
+#'
+#' @return null for expected complexity (linear), else throws error.
+#'
+#' @export
+
+expect_linear_time = function(f = asymptoticTimings, ...)
+{
+  timings.df <- f(...)
+
+  stopifnot(asymptoticTimeComplexityClass(timings.df) == "linear")
```

Comment:
stopifnot does not provide an informative error message so please use stop with an informative error message instead.

Reply:
yup right, I thought about using `stop(msg)` for the same reason but I forgot to change it!